injector/LogInjector.py

```
@@ -0,0 +1,158 @@
+        tree and injected the necessary logging statements. 
+
+        The NodeExtractor class is used to process an AST node and extract the 
+        necessary information. The SST class accepts a NodeExtractor object and 
+        uses the extracted information to populate the SST.
+    """
+    def __init__(self, sourcePath, ltCount):
+        self.sourcePath = sourcePath
+        self.sourceDir = os.path.dirname(sourcePath)
+        self.fileNameWExtension = os.path.basename(sourcePath)
+        self.fileName = pathlib.Path(sourcePath).stem
+        self.source = open(sourcePath,"r").read()
```

Ruff (0.8.2)

Use a context manager for opening files

(SIM115)